extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -167,38 +174,125 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory, JsonFacto
     }
     ContinuationData continuationData = new ContinuationData(nextPageData);
 
-    List<PhotoModel> photos = new ArrayList<>();
-    for (GoogleMediaItem mediaItem : mediaItemSearchResponse.getMediaItems()) {
-      if (mediaItem.getMediaMetadata().getPhoto() != null) {
-        // TODO: address videos later on
-        photos.add(new PhotoModel(
-            "", // TODO: no title?
-            mediaItem.getBaseUrl() + "=d",
-            mediaItem.getDescription(),
-            mediaItem.getMimeType(),
-            mediaItem.getId(),
-            albumId,
-            false));
-      }
+    PhotosContainerResource containerResource = null;
+    GoogleMediaItem[] mediaItems = mediaItemSearchResponse.getMediaItems();
+    if (mediaItems != null && mediaItems.length > 0) {
+      List<PhotoModel> photos = convertPhotosList(albumId, mediaItems, jobId);
+      containerResource = new PhotosContainerResource(null, photos);
     }
 
-    PhotosContainerResource containerResource = new PhotosContainerResource(null, photos);
-
     ResultType resultType = ResultType.CONTINUE;
     if (nextPageData == null) {
       resultType = ResultType.END;
     }
+
     return new ExportResult<>(resultType, containerResource, continuationData);
   }
 
+
+  @VisibleForTesting
+  void populateContainedPhotosList(UUID jobId, TokensAndUrlAuthData authData)
+      throws IOException {
+    // Get list of all album ids
+    List<String> albumIds = new LinkedList<>();
+    AlbumListResponse albumListResponse = getOrCreatePhotosInterface(authData)
+        .listAlbums(Optional.empty());
+    albumIds.addAll(Arrays.stream(albumListResponse.getAlbums()).map(GoogleAlbum::getId)
+        .collect(Collectors.toList()));
+    String token;
+    while ((token = albumListResponse.getNextPageToken()) != null) {
+      albumListResponse = getOrCreatePhotosInterface(authData)
+          .listAlbums(Optional.of(token));
+      albumIds.addAll(
+          Arrays.stream(albumListResponse
+              .getAlbums())
+              .map(GoogleAlbum::getId)
+              .collect(Collectors.toList()));
+    }
+
+    // Get list of all ids belonging to photos belonging to albums
+    TempPhotosData tempPhotosData = jobStore
```

Comment:
This seems confusing to me, why create it once and then update it bellow?

Isn't the logic that we just write this data once?  Why not construct the data to store and then just insert it?

Reply:
I think in general we want to pull the temp data and add to it, but since there won't be any temp data to start with for this case, it should be fine.